napari/_qt/widgets/qt_command_palette.py

```diff
@@ -0,0 +1,431 @@
+        self.setText(output_text)
+        return
+
+    def set_disabled(self) -> None:
+        """Set the label to disabled."""
+        text = self.command_text()
+        self.setText(colored(text, self.DISABLED_COLOR))
+        return
+
+
+class QCommandList(QtW.QListView):
+    commandClicked = Signal(int)  # one of the items is clicked
+
+    def __init__(self, parent: QtW.QWidget | None = None) -> None:
+        super().__init__(parent)
+        self.setCursor(Qt.CursorShape.PointingHandCursor)
+        self.setModel(QCommandMatchModel(self))
+        self.setSelectionMode(QtW.QAbstractItemView.SelectionMode.NoSelection)
+        self._selected_index = 0
+
+        # NOTE: maybe useful for fetch-and-scrolling in the future
+        self._index_offset = 0
+
+        self._label_widgets: list[QCommandLabel] = []
+        self._current_max_index = 0
+        for i in range(self.model()._max_matches):
+            lw = QCommandLabel()
+            self._label_widgets.append(lw)
+            self.setIndexWidget(self.model().index(i), lw)
+        self.pressed.connect(self._on_clicked)
+
+        self.setVerticalScrollBarPolicy(Qt.ScrollBarPolicy.ScrollBarAlwaysOff)
+
+        self._match_color = '#468cc6'
+        self._app_model_context: dict[str, Any] = {}
+
+    def _on_clicked(self, index: QtCore.QModelIndex) -> None:
+        if index.isValid():
+            self.commandClicked.emit(index.row())
+            return
+
+    def move_selection(self, dx: int) -> None:
+        """Move selection by dx, dx can be negative or positive."""
+        self._selected_index += dx
+        self._selected_index = max(0, self._selected_index)
+        self._selected_index = min(
+            self._current_max_index - 1, self._selected_index
+        )
+        self.update_selection()
+        return
+
+    def update_selection(self) -> None:
+        """Update the widget selection state based on the selected index."""
+        index = self.model().index(self._selected_index - self._index_offset)
+        if model := self.selectionModel():
+            model.setCurrentIndex(
+                index, QtCore.QItemSelectionModel.SelectionFlag.ClearAndSelect
+            )
+        return
+
+    @property
+    def all_commands(self) -> list[CommandRule]:
+        return self.model()._commands
+
+    def extend_command(self, commands: list[CommandRule]) -> None:
+        """Extend the list of commands."""
+        self.all_commands.extend(commands)
+        return
+
+    def command_at(self, index: int) -> CommandRule | None:
+        i = index - self._index_offset
+        index_widget = self.indexWidget(self.model().index(i))
+        if index_widget is None:
+            return None
+        return index_widget.command()
+
+    def iter_command(self) -> Iterator[CommandRule]:
+        """Iterate over all the commands registered to this command list widget."""
+        for i in range(self.model().rowCount()):
+            if not self.isRowHidden(i):
```

Comment:
If more than`max_matches` commands matched, list will be filled with visible rows. If not, only the texts of first rows are updated and other rows are hidden. Is this your question?

Reply:
Thanks. I guess I was wondering why we can't just set the number of rows to be the max of commands matched or 80, but I assume this is because I don't understand Qt or widgets :sweat_smile: